x/nft/approval_ops.go

```diff
@@ -0,0 +1,72 @@
+package nft
+
+import (
+	"errors"
+
+	"github.com/iov-one/weave"
+)
+
+type ApprovalOps struct {
+	owner weave.Address
+	//TODO: Possibly define a type for it, e.g. *ActionApprovalsSet
+	approvals *[]*ActionApprovals
+}
+
+//TODO: Sort errors and their codes
+func NewApprovalOps(nft Owned, approvals *[]*ActionApprovals) *ApprovalOps {
+	return &ApprovalOps{owner: nft.OwnerAddress(), approvals: approvals}
+}
+
+func (o *ApprovalOps) List() Approvals {
+	res := make(map[string]ApprovalMeta, 0)
+	for _, v := range *o.approvals {
+		res[v.Action] = v.Approvals
+	}
+	return res
+}
+
+func (o *ApprovalOps) Revoke(action string, from weave.Address) error {
+	if from == nil || from.Equals(o.owner) {
+		return errors.New("invalid account")
+	}
+	approvalsToRemove := o.List().ForAction(action).ForAddress(from)
+	if len(approvalsToRemove) == 0 {
+		return errors.New("does not exist")
+	}
+	for _, a := range approvalsToRemove {
+		for _, v := range a {
+			if v.Options.Immutable {
+				return errors.New("immutable and can not be changed")
+			}
+		}
+	}
+	*o.approvals = o.List().Filter(approvalsToRemove).AsPersistable()
+	return nil
+}
+
+//TODO: Note, that we are always excluding an owner from permissions, that might not be ideal
+//because upon transfer the owner might have some rights? Revisit
+//TODO: Figure out whether we need wildcard approvals, might be wise to add an ApprovalOptions flag
```

Comment:
I don't know if wild card approval is a good idea

Reply:
do you have a use case for wild card approval?

Reply:
@Isabello not for this one, but I supposed there could be a token that is exposing some of it's properties to the world, like, I don't know, a level or a map in game and such.

In theory that could be useful, but we can ignore that for now and leave the TODO